### ISSUE TITLE
Loss Function in Twenty_Newsgroups 

### ISSUE THREAD

**Issue by ghost** [cemoody/lda2vec#94]
I’m trying to adapt this example to my own tweets data, but I got really confused about the loss function calculation.
The calculation in the code is not identical to the formula in the paper.
![image](https://user-images.githubusercontent.com/8535587/53166298-eb4a4d00-35a2-11e9-96df-41f7c5030716.png)
![image](https://user-images.githubusercontent.com/8535587/53166307-f1d8c480-35a2-11e9-9f2f-145db5ffbc24.png)
The above two pictures show the Loss function formula in paper, but in the code, it only contains the first part of the sum, `L^d`. There is a dubious calculation of `l` underlined in the following picture, which I think might be the second part, but it’s not added to the loss before loss.backward().
![image](https://user-images.githubusercontent.com/8535587/53166346-0d43cf80-35a3-11e9-94c9-a450254f0783.png)
And I had to add a multiplication of `-clambda` to the original `prior` code (right under the red line in the above picture), because the returned value of model.prior() is actually a minus value of the formula (5) without the lambda factor. 
I don’t know Chainer, so it’s very hard to decide whether `l` was calculated correctly as the second part of the sum and could be readily added to the `loss` before `backward()`. It's also hard to decide the intention of the `fraction` variable and whether it should be multiplied to the `l` part of the sum too. Actually, I tried adding `l` to `loss` without multiplying it to `fraction` but didn’t get the expected result.

**Comment by ghost** [comment]
In the calculation of `l`:
![image](https://user-images.githubusercontent.com/8535587/53167165-2fd6e800-35a5-11e9-95dc-12102af2eefc.png)

Here above it seems to have set `self.sampler` to an instance of `NegativeSampling`.

![image](https://user-images.githubusercontent.com/8535587/53167306-922fe880-35a5-11e9-8ae9-c14dafefa464.png)

Here above in the `for` loop, `loss` was assigned the return of calling `NegativeSampling` each time. So does it discard the previous value each time while the formula in paper says it should be added together? Or some nature of the `self.sampler` made them accumulated?
